### PR TITLE
[Azure AI - Health Insights] - Typespec revision for Health Insights API Version 2

### PR DESCRIPTION
**Pull request by asaflevi-ms**
# Data Plane API - Pull Request
Health Insights (formaerly Health Decision Support) was released to Public Preview at the end of March 2023. We are working on a new API version which will include some new AI models in preview.

Base on private repo PR: https://github.com/Azure/azure-rest-api-specs-pr/pull/13119

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue: https://github.com/Azure/azure-rest-api-specs-pr/issues/12510

Is this review for (select one):

- [ ] a private preview
- [X] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/ai/HealthInsights
* Updated paths: Common, RadiologyInsights (New), PatientTimeline (New), OncoPhenotype, TrialMatcher

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

**Review by mikekistler — APPROVED** [review]
This looks fine for preview but I want to mention that we are close to finalizing an update to our LRO guidelines that addresses "non-resource oriented" LROs such as the ones in this API, and the new guidelines will recommend using PUT to initiate the (LRO) job, rather that POST as is done in this API.

**Comment by mikekistler** [comment]
I replaced @tg-msft with @JeffreyRichter in the reviewers list as Ted has stepped away from API Stewardship for a time and Jeff should review the LRO pattern here.

**Comment by mikekistler** [comment]
Link to API Review issue: https://github.com/Azure/azure-rest-api-specs-pr/issues/12510

**Comment by JeffreyRichter** [comment]
The Azure API Stewardship Board has recently modified our LRO guidance to include scenarios similar to what you appear to need, (we never had guidance on these scenarios before) see https://github.com/microsoft/api-guidelines/blob/f11b219cfcb846fab605e80ca72ae96477a412f5/azure/Guidelines.md#long-running-operations-lros.

Could you adopt this guidance and in particular this section: https://github.com/microsoft/api-guidelines/blob/f11b219cfcb846fab605e80ca72ae96477a412f5/azure/Guidelines.md#lro-action-no-resource

Do clients ever need to LIST the LRO operations that are in flight? Your current API does not support this, but the new guidance does offer a way of making this possible.

**Comment by asaflevi-ms** [comment]
> The Azure API Stewardship Board has recently modified our LRO guidance to include scenarios similar to what you appear to need, (we never had guidance on these scenarios before) see https://github.com/microsoft/api-guidelines/blob/f11b219cfcb846fab605e80ca72ae96477a412f5/azure/Guidelines.md#long-running-operations-lros.
> 
> Could you adopt this guidance and in particular this section: https://github.com/microsoft/api-guidelines/blob/f11b219cfcb846fab605e80ca72ae96477a412f5/azure/Guidelines.md#lro-action-no-resource
> 
> Do clients ever need to LIST the LRO operations that are in flight? Your current API does not support this, but the new guidance does offer a way of making this possible.

Hi @JeffreyRichter 
Thanks for you comment.
We are going to public preview next week and our GA is planned for end of Match, I prefer to adopt this change right after our public preview.

CC: @gils-ms

**Comment by mikekistler** [comment]
I’ve added the signedOff label so you can move forward with your preview.  -Mike

From: Jeffrey Richter ***@***.***>
Date: Wednesday, January 24, 2024 at 2:48 PM
To: Azure/azure-rest-api-specs ***@***.***>
Cc: Mike Kistler ***@***.***>, Review requested ***@***.***>
Subject: Re: [Azure/azure-rest-api-specs] [Azure AI - Health Insights] - Typespec revision for Health Insights API Version 2 (PR #27223)

The Azure API Stewardship Board has recently modified our LRO guidance to include scenarios similar to what you appear to need, (we never had guidance on these scenarios before) see https://github.com/microsoft/api-guidelines/blob/f11b219cfcb846fab605e80ca72ae96477a412f5/azure/Guidelines.md#long-running-operations-lros.
Could you adopt this guidance and in particular this section: https://github.com/microsoft/api-guidelines/blob/f11b219cfcb846fab605e80ca72ae96477a412f5/azure/Guidelines.md#lro-action-no-resource
Do clients ever need to LIST the LRO operations that are in flight? Your current API does not support this, but the new guidance does offer a way of making this possible.

Hi @JeffreyRichter<https://github.com/JeffreyRichter> Thanks for you comment. We are going to public preview next week and our GA is planned for end of Match, I prefer to adopt this change right after our public preview.

CC: @gils-ms<https://github.com/gils-ms>

It's fine to wait until after public preview.
I'm curious to know if the new guidelines would be a problem in terms of it working for your customers. So please don't wait too look to look at it (not implement it) to see if it would work well for your scenarios so we have time to address any issues or make tweaks to the guidance.

**Comment by mikeharder** [comment]
`AvoidAnonymousTypes` was supressed to workaround https://github.com/Azure/typespec-azure-pr/issues/3349

File: specification/ai/HealthInsights/HealthInsights.Common/model.common.fhir.elements.tsp
Comment (catalinaperalta) [inline specification/ai/HealthInsights/HealthInsights.Common/model.common.fhir.elements.tsp:60]: Projected name decorators should be in the client.tsp

File: specification/ai/HealthInsights/HealthInsights.OncoPhenotype/client.tsp
Comment (catalinaperalta) [inline specification/ai/HealthInsights/HealthInsights.OncoPhenotype/client.tsp:25]: If the FHIR models are not referenced from the Oncophenotype models, then these renames shouldnt be in this client.tsp. Same comment applies to the other client.tsp files for products that dont reference the FHIR models. Wont block the PR on this, since this is an SDK specific rename. Adding this tracking issue to follow up on whether these renames are necessary in all client.tsp files: https://github.com/Azure/azure-rest-api-specs/issues/27594

**Review by catalinaperalta — APPROVED** [review]

File: specification/ai/HealthInsights/main.tsp
Comment (weshaggard) [inline specification/ai/HealthInsights/main.tsp:1]: Is there a reason this was moved up into the parent folder as opposed to where it was under the `HealthInsights.OpenAPI` folder? Having nested projects can cause some difficulties in discoverability and processing.

Comment (catalinaperalta) [reply specification/ai/HealthInsights/main.tsp:1]: This is one service so having main.tsp at the root of the service directory is recommended since all of their products version together in the api.

File: specification/ai/HealthInsights/HealthInsights.OncoPhenotype/tspconfig.yaml
Comment (weshaggard) [inline specification/ai/HealthInsights/HealthInsights.OncoPhenotype/tspconfig.yaml:9]: Invalid path given the move which is failing your sdk generation step.

File: specification/ai/HealthInsights/HealthInsights.PatientTimeline/tspconfig.yaml
Comment (weshaggard) [inline specification/ai/HealthInsights/HealthInsights.PatientTimeline/tspconfig.yaml:9]: Invalid path given the move which is failing your sdk generation step.

File: specification/ai/HealthInsights/HealthInsights.RadiologyInsights/tspconfig.yaml
Comment (weshaggard) [inline specification/ai/HealthInsights/HealthInsights.RadiologyInsights/tspconfig.yaml:9]: Invalid path given the move which is failing your sdk generation step.

File: specification/ai/HealthInsights/HealthInsights.TrialMatcher/tspconfig.yaml
Comment (weshaggard) [inline specification/ai/HealthInsights/HealthInsights.TrialMatcher/tspconfig.yaml:9]: Invalid path given the move which is failing your sdk generation step.

File: specification/ai/HealthInsights/HealthInsights.OncoPhenotype/tspconfig.yaml
Comment (catalinaperalta) [reply specification/ai/HealthInsights/HealthInsights.OncoPhenotype/tspconfig.yaml]: Thanks for catching that, updated it for the team in all instances

**Comment by catalinaperalta** [comment]
NOTE: The following CI failures are pending fixes:
- TypeSpec APIView is currently checking in sub-directories and attempting to generate an APIView for each folder that has a tspconfig.yaml, instead we need to generate the TypeSpec APIView for the whole spec with the tspconfig.yaml file that configures the typespec-autorest emitter. Tracking issue: https://github.com/Azure/azure-sdk-tools/issues/7599
- The SDK auto-generation steps are currently failing due to not considering auto-generating different products under 1 service directory. Tracking issue: https://github.com/Azure/azure-sdk-tools/issues/7603

**Comment by raych1** [comment]
@weidongxu-microsoft , Java sdk failed in generation step but doesn't have any details for the error when I verified locally. Do you know what caused the failure?
```
npx tsp compile /home/raychen/repos/tmp/work-dir/azure-sdk-for-java/sdk/healthinsights/azure-healthinsights-cancerprofiling/TempTypeSpecFiles/HealthInsights.OncoPhenotype/client.tsp --emit @azure-tools/typespec-java --option @azure-tools/typespec-java.emitter-output-dir=/home/raychen/repos/tmp/work-dir/azure-sdk-for-java/sdk/healthinsights/azure-healthinsights-cancerprofiling/
Write-Error: Failed to generate sdk project at /home/raychen/repos/tmp/work-dir/azure-sdk-for-java/sdk/healthinsights/azure-healthinsights-cancerprofiling
```

**Comment by raych1** [comment]
> * The SDK auto-generation steps are currently failing due to not considering auto-generating different products under 1 service directory. Tracking issue: [[SDK automation] SDK automation for TypeSpec PRs in the rest repo need to consider multiple products under a service directory azure-sdk-tools#7603](https://github.com/Azure/azure-sdk-tools/issues/7603)

@catalinaperalta , sdk automation supports multiple TypeSpec projects scenario.  For instance, see this [dotnet run logs](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3453602&view=logs&j=03afb3bb-7296-55ad-aa07-ceee610c73b2&t=0f320c29-1c90-5304-d2d3-c58a6bb76437). All the four packages failed to be generated. 
//CC: @weshaggard 

+@chunyu3 , can you  troubleshoot if the generation failures are caused by spec issue?
```
"packages": [
    {
      "language": ".Net",
      "changelog": {
        "content": null,
        "hasBreakingChange": null
      },
      "result": "failed",
      "packageName": "Azure.Health.Insights.CancerProfiling",
      "packageFolder": "/mnt/vss/_work/1/s/azure-sdk-for-net/sdk/healthinsights/Azure.Health.Insights.CancerProfiling",
      "path": [
        "./sdk/healthinsights/Azure.Health.Insights.CancerProfiling",
        "sdk/healthinsights/ci.yml"
      ],
      "artifacts": [],
      "installInstructions": {
        "full": null,
        "lite": null
      },
      "apiViewArtifact": ""
    },
    {
      "path": [
        ""
      ],
      "result": "failed"
    },
    {
      "path": [
        ""
      ],
      "result": "failed"
    },
    {
      "path": [
        ""
      ],
      "result": "failed"
    }
  ]
```

**Comment by chunyu3** [comment]
> > * The SDK auto-generation steps are currently failing due to not considering auto-generating different products under 1 service directory. Tracking issue: [[SDK automation] SDK automation for TypeSpec PRs in the rest repo need to consider multiple products under a service directory azure-sdk-tools#7603](https://github.com/Azure/azure-sdk-tools/issues/7603)
> 
> @catalinaperalta , sdk automation supports multiple TypeSpec projects scenario. For instance, see this [dotnet run logs](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3453602&view=logs&j=03afb3bb-7296-55ad-aa07-ceee610c73b2&t=0f320c29-1c90-5304-d2d3-c58a6bb76437). All the four packages failed to be generated. //CC: @weshaggard
> 
> +@chunyu3 , can you troubleshoot if the generation failures are caused by spec issue?
> 
> ```
> "packages": [
>     {
>       "language": ".Net",
>       "changelog": {
>         "content": null,
>         "hasBreakingChange": null
>       },
>       "result": "failed",
>       "packageName": "Azure.Health.Insights.CancerProfiling",
>       "packageFolder": "/mnt/vss/_work/1/s/azure-sdk-for-net/sdk/healthinsights/Azure.Health.Insights.CancerProfiling",
>       "path": [
>         "./sdk/healthinsights/Azure.Health.Insights.CancerProfiling",
>         "sdk/healthinsights/ci.yml"
>       ],
>       "artifacts": [],
>       "installInstructions": {
>         "full": null,
>         "lite": null
>       },
>       "apiViewArtifact": ""
>     },
>     {
>       "path": [
>         ""
>       ],
>       "result": "failed"
>     },
>     {
>       "path": [
>         ""
>       ],
>       "result": "failed"
>     },
>     {
>       "path": [
>         ""
>       ],
>       "result": "failed"
>     }
>   ]
> ```

The error is
image

It because you put the `service.tsp` at the top level directory and the tsp-location.yaml does not include the top-level directory, so it does not sync this file to generate.  I suggest to put this common tsp in a directory called `***.common` or something, then add the common directory in the `additionalDirectories` in the tsp-location.yaml

**Comment by weshaggard** [comment]
Thanks @chunyu3.

@asaflevi-ms @catalinaperalta we should consider moving those back into their own sub directory or moving them under the common project and then add whichever path back to the additionalDirectories in the tspconfig.yaml files.

**Comment by catalinaperalta** [comment]
Follow up about the remaining CI failures [NOT blocking PR merge]:
- TypeSpec APIView: the check will be updated to account for projects with similar structures.
  - Tracking issue: https://github.com/Azure/azure-sdk-tools/issues/7599
- SDK azure-sdk-for-net-track2: 
  - There are some test failures in the SDK due to some renames in the spec. @asaflevi-ms please take a look at these issues and update in the .NET SDK tests where necessary before your next release.
  - There is an issue with using a namespace that is not approved. @asaflevi-ms and I will follow up on this.
- SDK azure-sdk-for-java:
  - @raych1 @weidongxu-microsoft looks like a tool failure to find the package to build, please help investigate. 
- SDK azure-sdk-for-js: JS SDK build succeeded but the PublishTestResults step failed.
  - @raych1 could you also please help investigate why the PublishTestResults failed.

Thanks @weshaggard for helping investigate and catching the issues in the PR!

**Comment by weidongxu-microsoft** [comment]
Could be a bug in Java automation, when one PR involves changes to multiple packages (new and existing). Will check on the PR migrating tsp-client.

**Comment by raych1** [comment]
@catalinaperalta , for SDK azure-sdk-for-js, it's a known issue in the `PublishTestResults` step which is tracked [here](https://github.com/Azure/azure-sdk-tools/issues/7458)

File: specification/ai/HealthInsights/HealthInsights.RadiologyInsights/tspconfig.yaml
Comment (weidongxu-microsoft) [inline specification/ai/HealthInsights/HealthInsights.RadiologyInsights/tspconfig.yaml:36]: The error on `package-dir`. It shoudl be `azure-health-insights-radiologyinsights` (see the pattern in namespace)

**Pull request merged**
